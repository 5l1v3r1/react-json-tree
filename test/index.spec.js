// @flow
import React from 'react';
import test from 'ava';
import renderer from 'react-test-renderer';

import JSONTree from '../src/index';

const BASIC_DATA = { a: 1, b: 'c' };

test('should render basic tree', t => {
  t.snapshot(renderer.create(<JSONTree data={BASIC_DATA} />).toJSON());
});
