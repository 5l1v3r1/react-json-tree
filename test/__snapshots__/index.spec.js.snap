exports[`should render basic tree 1`] = `"{\"__ava_react_jsx\":{\"type\":\"ul\",\"props\":{\"style\":{\"border\":0,\"padding\":0,\"marginTop\":\"0.5em\",\"marginBottom\":\"0.5em\",\"marginLeft\":\"0.125em\",\"marginRight\":0,\"listStyle\":\"none\",\"MozUserSelect\":\"none\",\"WebkitUserSelect\":\"none\",\"backgroundColor\":\"#ffffff\"}},\"children\":[{\"type\":\"li\",\"props\":{\"style\":{\"position\":\"relative\",\"paddingTop\":\"0.25em\",\"marginLeft\":0,\"paddingLeft\":0}},\"children\":[{\"type\":\"div\",\"props\":{\"style\":{\"display\":\"inline-block\",\"paddingRight\":\"0.5em\",\"paddingLeft\":0,\"cursor\":\"pointer\"}},\"children\":[{\"type\":\"div\",\"props\":{\"style\":{\"marginLeft\":0,\"transition\":\"150ms\",\"WebkitTransition\":\"150ms\",\"MozTransition\":\"150ms\",\"WebkitTransform\":\"rotateZ(90deg)\",\"MozTransform\":\"rotateZ(90deg)\",\"transform\":\"rotateZ(90deg)\",\"transformOrigin\":\"45% 50%\",\"WebkitTransformOrigin\":\"45% 50%\",\"MozTransformOrigin\":\"45% 50%\",\"position\":\"relative\",\"lineHeight\":\"1.1em\",\"fontSize\":\"0.75em\",\"color\":\"#2287cd\"}},\"children\":[\"▶\"]}]},{\"type\":\"label\",\"props\":{\"style\":{\"display\":\"inline-block\",\"color\":\"#2287cd\",\"margin\":0,\"padding\":0,\"WebkitUserSelect\":\"inherit\",\"MozUserSelect\":\"inherit\",\"cursor\":\"pointer\"}},\"children\":[{\"type\":\"span\",\"props\":{},\"children\":[\"root\",\":\"]}]},{\"type\":\"span\",\"props\":{\"style\":{\"paddingLeft\":\"0.5em\",\"cursor\":\"default\",\"color\":\"#60767e\"}},\"children\":[{\"type\":\"span\",\"props\":{},\"children\":[\"{}\",\" · 2 keys\"]}]},{\"type\":\"ul\",\"props\":{\"style\":{\"padding\":0,\"margin\":0,\"listStyle\":\"none\",\"display\":\"block\"}},\"children\":[{\"type\":\"li\",\"props\":{\"style\":{\"paddingTop\":\"0.25em\",\"paddingRight\":0,\"marginLeft\":\"0.875em\",\"WebkitUserSelect\":\"text\",\"MozUserSelect\":\"text\",\"wordWrap\":\"break-word\",\"paddingLeft\":\"2.125em\",\"textIndent\":\"-0.5em\",\"wordBreak\":\"break-all\",\"whiteSpace\":\"pre-wrap\"}},\"children\":[{\"type\":\"label\",\"props\":{\"style\":{\"display\":\"inline-block\",\"color\":\"#2287cd\",\"margin\":\"0 0.5em 0 0\"}},\"children\":[{\"type\":\"span\",\"props\":{},\"children\":[\"a\",\":\"]}]},{\"type\":\"span\",\"props\":{\"style\":{\"color\":\"#da5a25\"}},\"children\":[1]}]},{\"type\":\"li\",\"props\":{\"style\":{\"paddingTop\":\"0.25em\",\"paddingRight\":0,\"marginLeft\":\"0.875em\",\"WebkitUserSelect\":\"text\",\"MozUserSelect\":\"text\",\"wordWrap\":\"break-word\",\"paddingLeft\":\"2.125em\",\"textIndent\":\"-0.5em\",\"wordBreak\":\"break-all\",\"whiteSpace\":\"pre-wrap\"}},\"children\":[{\"type\":\"label\",\"props\":{\"style\":{\"display\":\"inline-block\",\"color\":\"#2287cd\",\"margin\":\"0 0.5em 0 0\"}},\"children\":[{\"type\":\"span\",\"props\":{},\"children\":[\"b\",\":\"]}]},{\"type\":\"span\",\"props\":{\"style\":{\"color\":\"#9aae16\"}},\"children\":[\"\\\"c\\\"\"]}]}]}]}]}}"`;
